UDPServer/ViewModels/ShellViewModel.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Caliburn.Micro;
+using Sice.PoC.UDPServer;
+using System.Collections.ObjectModel;
+using System.Configuration;
+using System.Net;
+using System.Windows.Media;
+
+namespace UDPServer.ViewModels
+{
+    public class ShellViewModel : Screen, IHandle<ReceivedMessage>, IHandle<SiceUDPServer.ServerStatus>
+    {
+        private readonly IEventAggregator _eventAggregator;
+        private readonly SiceUDPServer server;
+        public ShellViewModel(IEventAggregator eventAggregator, SiceUDPServer siceUDPServer)
+        {
+            this.IP = ConfigurationManager.AppSettings["IPAddress"];
+            this.Port = ConfigurationManager.AppSettings["PortNumber"];
+            this.server = siceUDPServer;
+            this.ReceivedMessages = new ObservableCollection<string>();
+            this._eventAggregator = eventAggregator;
+            this._eventAggregator.Subscribe(this);
+        }
+
+        private string _iP;
+        private string _port;
+        private string _listeningStatus;
+        private SolidColorBrush _statusTextColour;
+
+        public string IP
+        {
+            get { return _iP; }
+            set {
+                _iP = value;
+                NotifyOfPropertyChange(() => IP);
```

Comment:
use the NotifyOfPropertyChange(nameof(...)) instead as it's more performant.

Reply:
Changed